### PR TITLE
fix: improve kafka bindings

### LINKED ISSUE #673: Update Kafka Adapter to support spec v3
#### Describe the bug
We need to update the kafka adapter to support v3 of the spec. Currently, from what I noticed, the parsed methods seem to be different from v2 and it causes the adapter to fail. 

#### How to Reproduce
Update the spec to v3 and run the project in the `examples/kafka-test` directory. First, you'll get an error related to the `reply.channel`(I'll have to check more, I'm slightly confused by the reply feature). However, we can get past that by adding a conditional to check for `reply`. Finally once the `reply.channel` error out of the way, you'll get the following error

`TypeError: Cannot read properties of undefined (reading 'type')
    at file:///Users/a/Documents/asyncapi/glee/dist/adapters/kafka/index.js:29:87
    at Array.find (<anonymous>)
    at KafkaAdapter.<anonymous> (file:///Users/a/Documents/asyncapi/glee/dist/adapters/kafka/index.js:29:69)
    at Generator.next (<anonymous>)
    at fulfilled (file:///Users/a/Documents/asyncapi/glee/dist/adapters/kafka/index.js:4:58) `

#### Expected behavior
The Kafka adapter should start correctly without errors.

@Souvikns @KhudaDad414 I'll start working on a fix as this is `blocking #411 `.

### PR DESCRIPTION
**Pull request by oviecodes**
**Description**

This PR contains a fix for the Kafka adapter. The Kafka adapter should now start properly and run well.

Kindly have a look @Souvikns @KhudaDad414 , we also need to discuss about the `setUpReplyMiddlewares` in the `src/index.js` file.

**Related issue(s)**
`Fixes #673 `

### CODE REVIEW

File: src/adapters/kafka/index.ts [hunk 1]
```diff
@@ -15,20 +15,17 @@ class KafkaAdapter extends Adapter {
...
-      (sec) => {
-        const secName = Object.keys(sec.values())[0]
-        return this.parsedAsyncAPI.components().securitySchemes().get(secName)
-      }
+    const securityRequirements = this.AsyncAPIServer.json().security.map(
+      (sec) => sec
     )
     const userAndPasswordSecurityReq = securityRequirements.find(
-      (sec) => sec.type() === 'userPassword'
+      (sec) => sec.type === 'userPassword'
```
Comment (KhudaDad414) [inline src/adapters/kafka/index.ts:22]: @oviecodes what do you think about extracting these constants inside an enum? then reference it here as something like `SECURITY_TYPE.USER_PASSWORD` or `SECURITY_TYPE.SCRAM_SHA_512` 🤔

Comment (oviecodes) [reply src/adapters/kafka/index.ts]: alright @KhudaDad414 .

**Comment by KhudaDad414** [comment]
@oviecodes the branch has some conflicts.

**Comment by Souvikns** [comment]
@oviecodes you have a lot of conflicts, do you need any help?

**Comment by oviecodes** [comment]
i'll fix it @Souvikns , I've been a little occupied lately.

**Comment by KhudaDad414** [comment]
@oviecodes This PR needs a little love, if you have the time of course.

**Comment by oviecodes** [comment]
I'll fix this @KhudaDad414 ... sorry guys

**Comment by oviecodes** [comment]
@KhudaDad414 could you check this out, so I can continue with the main issue

File: docs/pages/env-vars-config.md
Comment (KhudaDad414) [inline docs/pages/env-vars-config.md:1]: Can you revert the changes to the docs? since this PR is about KAFKA bindings I don't think it's appropriate for it to include any other changes.

**Review by KhudaDad414 — CHANGES_REQUESTED** [review]

File: docs/pages/env-vars-config.md
Comment (oviecodes) [reply docs/pages/env-vars-config.md:1]: this PR is old, I had to merge it with the current master branch, so there isn't any change that's not already on the master branch ...

Comment (KhudaDad414) [reply docs/pages/env-vars-config.md:1]: If Diff shows there are, then there are changes that this PR is proposing. 
for example. master doesn't have this file: `docs/pages/glee-template.md` since it was removed by a previous PR.
but you are adding it again.

Comment (oviecodes) [reply docs/pages/env-vars-config.md:1]: oh Damn, I'm just pulling from the master branch. I'll try updating from master again, if it doesn't work then I'll scrap this PR, it seems it's too old and things are getting a bit more complicated.

Comment (KhudaDad414) [reply docs/pages/env-vars-config.md:1]: @oviecodes yep. I think the problem is that you are pulling from your fork's master instead of this repo's master. happened to me multiple times. 😆

**Comment by oviecodes** [comment]
@KhudaDad414 I'll open another pull. request, I'll close this one

**Comment by oviecodes** [comment]
@KhudaDad414  I'll open another pull. request, I'll close this one
